src/Cmd/Extra.elm

```diff
@@ -41,12 +64,111 @@ perform =
 
 
 {-| Similar to perform but takes `Result msg` and performs action only on `Ok`.
-```
-attempt (Ok 1) : Cmd number
-attempt (Ok "I'm fine") : Cmd String
-attempt (Err "Failed") == Cmd.none => True
-```
+
+    attempted : Cmd String
+    attempted =
+      attempt <| Ok "I'm fine"
+
+    attempt (Err "Failed") == Cmd.none
+    --> True
+
 -}
 attempt : Result x msg -> Cmd msg
 attempt =
     Result.withDefault Cmd.none << Result.map perform
+
+
+{-| Similar to attempt byt works with `Maybe` instead
+
+    maybeCmd : Cmd String
+    maybeCmd =
+        maybe <| Just 1
+
+    maybe Nothing == Cmd.none
+    --> True
+
+-}
+maybe : Maybe msg -> Cmd msg
+maybe maybe =
+    Maybe.map perform maybe
+        |> Maybe.withDefault Cmd.none
+
+
+{-| Construct from Result.
+
+    resultCmd : Cmd String
+    resultCmd =
+      fromResult toString (Ok 1)
+
+    fromResult toString (Err ())
+    --> Cmd.none
+
+-}
+fromResult : (a -> msg) -> Result x a -> Cmd msg
+fromResult const res =
+    Result.map const res
+        |> attempt
+
+
+{-| Construct from Maybe.
+
+    stringCmd : Cmd String
+    stringCmd =
+      toString (Just 1)
+
+    fromMaybe toString Nothing
+    --> Cmd.none
+
+-}
+fromMaybe : (a -> msg) -> Maybe a -> Cmd msg
+fromMaybe const m =
+    Maybe.map const m
+        |> maybe
+
+
+
+-- Chaining in update
+
+
+{-| Creates pair `model` with `Cmd.none`
+
+    pair : ( String, Cmd msg )
+    pair = pure "foo"
+
+    pair
+      |> Tuple.second
+      |> ((==) Cmd.none)
+    --> True
+
+-}
+pure : model -> ( model, Cmd msg )
+pure model =
+    ( model, Cmd.none )
+
+
+{-| Add Cmd to model to create pair.
+-}
+with : Cmd msg -> model -> ( model, Cmd msg )
+with cmd model =
+    ( model, cmd )
+
+
+{-| Add new cmd to existing pair.
+-}
+add : Cmd msg -> ( model, Cmd msg ) -> ( model, Cmd msg )
```

Comment:
Question: Wouldn't that be a problem when you chain multiple `add` together rather than do one `Cmd.batch[...]` vs `Cmd.batch [cmd, Cmd.batch [cmd, Cmd.batch [cmd, ...]]]`.

Reply:
nope. It's a common pattern we're already using https://github.com/turboMaCk/glue/blob/master/src/Glue.elm#L215-L221

Reply:
OK then.